data/mods/gen2/statuses.js

```diff
@@ -226,6 +223,31 @@ let BattleStatuses = {
 			this.effectData.duration = 2;
 		},
 	},
+	residualdmg: {
+		name: 'residualdmg',
+		id: 'residualdmg',
+		num: 0,
+		onStart(target) {
+			target.volatiles['residualdmg'].counter = 0;
+		},
+		onAfterMoveSelfPriority: 100,
+		onAfterMoveSelf(pokemon) {
+			if (['brn', 'psn', 'tox'].includes(pokemon.status)) pokemon.volatiles['residualdmg'].counter++;
+		},
```

Comment:
I think the last thing to do then would be increasing the counter during `onAfterSwitchInSelf` here as well. Currently 3 turns of Toxic Baton Passing into an already burned Pokemon keeps the counter at 3 for the switch turn.

Reply:
Done!